docs/sources/developers/plugins/add-support-for-annotations.md

```diff
@@ -8,140 +8,26 @@ This guide explains how to add support for [annotations]({{< relref "../../dashb
-  const events: AnnotationEvent[] = [];
-
-  const date = new Date();
-
-  const event: AnnotationEvent = {
-    time: date.valueOf(),
-    text: 'foo',
-    tags: ['bar'],
-  };
-
-  events.push(event);
-
-  return events;
-}
-```
-
-## Region annotations
-
-[Region annotations]({{< relref "../../dashboards/annotations.md#adding-regions-events" >}}) have a start and end time. This can for example be used to annotate maintenance windows or downtime.
-
-To return a region annotation, set the `timeEnd`, and `isRegion` properties.
+To enable annotation support for your data source, add the following two lines of code. Grafana uses your default query editor for editing annotation queries.
 
-```ts
-const regionEvent: AnnotationEvent = {
-  time: startDate.valueOf(),
-  timeEnd: endDate.valueOf(),
-  isRegion: true,
-  text: 'foo',
-  tags: ['bar'],
-};
-```
+1. Add `"annotations": true` to the [plugin.json]({{< relref "metadata.md" >}}) file.
 
-## Build a annotation query editor
+   **plugin.json**
 
-Let users write custom annotation queries to only display the annotation events they care about, by adding a _query editor_. You only need to build a query editor if you want to let users query or filter annotations.
-
-> **Note:** Annotation query editors have yet to receive support for React. The instructions here are given for Angular. Fortunately, you can run Angular even in a plugin otherwise written using React. This section will be updated once React support for annotation queries editors is available.
-
-1. Create a file called `AnnotationQueryEditor.ts` in the `src` directory, with the following content.
-
-   ```ts
-   export class AnnotationQueryEditor {
-     static templateUrl = 'partials/annotations.editor.html';
-
-     annotation: any;
-
-     constructor() {
-       this.annotation.queryText = this.annotation.queryText || '';
-     }
+   ```json
+   {
+     "annotations": true
    }
    ```
 
-1. Create a directory called `partials` in the `src` directory.
-
-1. Create a file called `annotations.editor.html` in the `partials` directory you just created, with the following content.
-
-   ```html
-   <div class="gf-form-group">
-     <div class="gf-form-inline">
-       <div class="gf-form gf-form--grow">
-         <input
-          class="gf-form-input"
-          placeholder="query expression"
-          ng-model="ctrl.annotation.queryText"
-          ></input>
-       </div>
-     </div>
-   </div>
-   ```
-
-1. In your data source query—the one that extends [DataQuery]({{< relref "../../packages_api/data/dataquery.md" >}})—add the `queryText` property. The name of the property needs to correspond to the text in `ng-model`, e.g. `ctrl.annotation.<propertyName>`.
-
-   ```ts
-   export interface MyQuery extends DataQuery {
-     // ...
-     queryText?: string;
-   }
-   ```
+2. In `datasource.ts`, override the `annotations` method from `DataSourceApi`. For the default behavior, you can set `annotations` to an empty object.
```

Comment:
This seems pretty scarce on information and when comparing the old docs with new one it seems like something is missing. My assumption is that right now `DataSourceApi` has some default code so you don't need to explicitly code the annotations. At the same time:

> For the default behavior, you can set `annotations` to an empty object

No idea what this means. Like instead of method `annotations` property of the datasource should be an object? Or the method should return an empty object? Like none of those makes sense to me as a programmer.

I would expect something like, "DataSourceApi has a default implementation of annotations method that should be sufficient so no change is needed to the data source. If you want special behaviour you can override the annotations method." and then some information how that method should look like and what should it return.

Reply:
The API for this is a bit unintuitive tbh. Your first guess is correct though: `annotations` is an object, and not a method. You don't need to explicitly return the annotations anymore. By adding `annotations: {}`, you'll use the default query editor and the user can select the fields in the resulting data frame to use for each annotation property.

You do need to add `annotations: {}` to your data source. If you do, you'll get the default implementation.

Here's an example: 

https://github.com/marcusolsson/grafana-json-datasource/blob/4352df6d0a48dcbf14597d575a7cbc9f79a2a590/src/datasource.ts#L58-L61

Reply:
The reason why the previous docs had more instructions was mainly because you had to implement the annotation query, and a separate query editor. With the new implementation, you're using the same query editor as in the dashboard, and the `query` method from the DataSourceApi.

Users can then select the fields to use themselves:

![Screenshot 2021-03-09 at 11 18 28@2x](https://user-images.githubusercontent.com/8396880/110456015-610dac80-80c9-11eb-9adc-dbff9ee24034.png)

Reply:
I guess
> Your first guess is correct though: annotations is an object, and not a method
> In `datasource.ts`, override the `annotations` method from `DataSourceApi`.

is confusing. That just does not seem logically correct.

Reply:
Also what will happen if I don't add the line `annotations: {}`? I mean as far as default values go I would assume not adding it should also trigger some default behaviour.

Reply:
Oh, I see now. Yes, the docs should say 'object' rather than 'method'. I'll update this.

If you set `annotations: true`, the data source shows up in the list of data sources, but unless you add the `annotations: {}` lines, the query editor won't show up at all.

<img width="751" alt="Screenshot 2021-03-09 at 11 49 07@2x" src="https://user-images.githubusercontent.com/8396880/110459621-7c7ab680-80cd-11eb-879b-1337a29d05f8.png">